mattermost/mattermost-plugin-legal-hold
# Properly update and process hashes file

#### Summary

This fixes two issues:

- First one in the processor: The file being opened wasn't the one that required hashing. (introduced in #101)
- Second in the job system: The hash file was being used append only, meaning that we get a new `path,hash` file for the same file every loop of the processor. Now we open the file, check if the path exists and update its hash.
- **Breaking**: Changed the hashes file type from CSV to JSON.

#### How to test

##### Basic test scenario

1. Deploy this version of the plugin
2. Create a legal hold that contains users that have posted data in the past
3. Run the legal hold job manually by using #43 (wait for process to finish depending on dataset, check logs)
4. Download the legal hold bundle for this entry
5. Run the processor over it with the `--legal-hold-secret` flag set. You can retrieve the secret from the plugin settings in the system console, the button to the left of the download button.
6. Ensure file verification steps are happening
